Emergence/Contexts/Networking/ArtsyAPI.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+import RxSwift
+import Moya
+import Alamofire
+
+enum ArtsyAPI {
+    case XApp
+    case ShowInfo(showID:String)
+
+}
+
+extension ArtsyAPI : MoyaTarget {
+
+    var base: String { return AppSetup.sharedState.useStaging ? "https://stagingapi.artsy.net" : "https://api.artsy.net" }
+    var baseURL: NSURL { return NSURL(string: base)! }
+
+    var parameters: [String: AnyObject] {
+        switch self {
+
+        case .XApp:
+            return [
+                "grant_type": "credentials"
+            ]
+
+        default:
+            return [:]
+        }
+    }
+
+    var method: Moya.Method {
+        switch self {
+        default:
+            return .GET
+        }
+    }
+
+    // Only kinda guilty. Deal with it.
+    var sampleData: NSData {
+        switch self {
+
+        default:
+            return NSData()
+
+        }
+    }
+
+    var path: String {
+        switch self {
+
+        case .XApp:
+            return "/api/v1/xapp_token"
+
+        case .ShowInfo(let showID):
+            return "/api/v1/show/\(showID)"
+        }
+    }
+
+}
+
+// MARK: - Provider setup
+
+class ArtsyProvider<T where T: MoyaTarget> : RxMoyaProvider<T> {
+    var authToken: XAppToken
+
+    override init(endpointClosure: MoyaEndpointsClosure = MoyaProvider.DefaultEndpointMapping,
+                 endpointResolver: MoyaEndpointResolution = ArtsyProvider.AuthEndpointResolution,
+                     stubBehavior: MoyaStubbedBehavior = MoyaProvider.NoStubbingBehavior,
+                credentialClosure: MoyaCredentialClosure? = nil,
+           networkActivityClosure: Moya.NetworkActivityClosure? = nil,
+                          manager: Alamofire.Manager = Alamofire.Manager.sharedInstance) {
```

Comment:
and for a long time I was really confused about what was the class and what was the default behaviour, causing this: https://twitter.com/orta/status/648479879814643712